src/controllers/DynamicFacetRangeQueryController.ts

```diff
@@ -12,19 +12,23 @@ export class DynamicFacetRangeQueryController extends DynamicFacetQueryControlle
       currentValues: this.currentValues,
       numberOfValues: this.numberOfValues,
       freezeCurrentValues: false,
-      generateAutomaticRanges: !this.isRangesOptionDefined
+      generateAutomaticRanges: this.shouldGenerateAutomaticRanges
     };
   }
 
   protected get numberOfValues() {
-    return this.isRangesOptionDefined ? this.facet.options.ranges.length : this.facet.options.numberOfValues;
+    return this.shouldGenerateAutomaticRanges ? this.facet.options.numberOfValues : this.facet.options.ranges.length;
   }
 
-  private get isRangesOptionDefined() {
-    return !!this.facet.options.ranges.length;
+  private get shouldGenerateAutomaticRanges() {
+    return !this.facet.options.ranges.length;
```

Comment:
I find function and getter names are clearer when they describe what they do/hold, rather than how that information is later used. The old name captured this, whereas the new one is more abstract.

The conditions on line 20 and 28 become clearer imo when written as:
```
this.manualRangesAreDefined ? this.facet.options.ranges.length : this.facet.options.numberOfValues
```
```
if (!this.manualRangesAreDefined && !this.facet.hasActiveValues) {
    return []
}
```